src/chainstate/stacks/boot/pox_2_tests.rs

```diff
@@ -0,0 +1,322 @@
+            |ref mut miner,
+             ref mut sortdb,
+             ref mut chainstate,
+             vrf_proof,
+             ref parent_opt,
+             ref parent_microblock_header_opt| {
+                let parent_tip = get_parent_tip(parent_opt, chainstate, sortdb);
+                let coinbase_tx = make_coinbase(miner, tenure_id as usize);
+
+                let mut block_txs = vec![coinbase_tx];
+
+                if tenure_id == 1 {
+                    // Alice locks up exactly 25% of the liquid STX supply, so this should succeed.
+                    // this locks for cycles 6, 7, 8, 9.
+                    //  however, the v2 unlock occurs before cycle 8, so alice will be unlocked before cycles 8 and 9
+                    let alice_lockup = make_pox_lockup(
+                        &alice,
+                        0,
+                        1024 * POX_THRESHOLD_STEPS_USTX,
+                        AddressHashMode::SerializeP2PKH,
+                        key_to_stacks_addr(&alice).bytes,
+                        4,
+                        tip.block_height,
+                    );
+                    block_txs.push(alice_lockup);
+                } else if tenure_id == 10 {
+                    // Lets have Bob lock up for v2
+                    // this will lock for cycles 8, 9, 10, and 11
+                    //  the first v2 cycle will be 8
+                    let bob_lockup = make_pox_2_lockup(
+                        &bob,
+                        0,
+                        512 * POX_THRESHOLD_STEPS_USTX,
+                        AddressHashMode::SerializeP2PKH,
+                        key_to_stacks_addr(&bob).bytes,
+                        6,
+                        tip.block_height,
+                    );
+                    block_txs.push(bob_lockup);
+                } else if tenure_id == 11 {
+                    // Bob tries to lock in PoX v1 too, but it shouldn't work!
+                    let bob_lockup = make_pox_lockup(
+                        &bob,
+                        1,
+                        512 * POX_THRESHOLD_STEPS_USTX,
+                        AddressHashMode::SerializeP2PKH,
+                        key_to_stacks_addr(&bob).bytes,
+                        4,
+                        tip.block_height,
+                    );
+                    block_txs.push(bob_lockup);
+                } else if tenure_id == 13 {
+                    // Alice auto-unlocked, so stack in PoX v2
+                    let alice_lockup = make_pox_2_lockup(
+                        &alice,
+                        1,
+                        512 * POX_THRESHOLD_STEPS_USTX,
+                        AddressHashMode::SerializeP2PKH,
+                        key_to_stacks_addr(&alice).bytes,
+                        12,
+                        tip.block_height,
+                    );
+                    block_txs.push(alice_lockup);
+                } else if tenure_id == 31 {
```

Comment:
Is this condition reachable?  It looks like `tenure_id` maxes out at 25?

Reply:
This test does 35 tenures